src/command_modules/azure-cli-backup/azure/cli/command_modules/backup/_validators.py

```diff
@@ -0,0 +1,18 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from datetime import datetime
+
+# Argument types
+
+def datetime_type(string):
+    """ Validates UTC datettime in accepted format. Examples: 31-12-2017 """
+    accepted_date_formats = ['%d-%m-%Y', '%I:%M:%S', '%I:%M:%S.%f']
+    for form in accepted_date_formats:
+        try:
+            return datetime.strptime(string, form)
+        except ValueError:  # checks next format
+            pass
+    raise ValueError("Input '{}' not valid. Valid example: 31-12-2017".format(string))
```

Comment:
Can I know what is this for? To process user input or service output? If service output, I think the format should be fixed and the CLI should not "guess" the format.

Reply:
Updated this file. This is for processing user input and not service input.